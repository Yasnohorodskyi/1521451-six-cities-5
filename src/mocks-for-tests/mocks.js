export const offers = [
  {
    city: {
      name: `Hamburg`,
      location: {
        latitude: 53.550343,
        longitude: 10.000654,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`
    ],
    title: `The house among olive `,
    isFavorite: true,
    isPremium: false,
    rating: 4.3,
    type: `house`,
    bedrooms: 5,
    maxAdults: 6,
    price: 190,
    goods: [
      `Washer`,
      `Breakfast`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    location: {
      latitude: 53.53434,
      longitude: 9.998654,
      zoom: 16
    },
    id: 11
  },
  {
    city: {
      name: `Paris`,
      location: {
        latitude: 48.85661,
        longitude: 2.351499,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`
    ],
    title: `Tile House`,
    isFavorite: false,
    rating: 2.9,
    type: `hotel`,
    bedrooms: 2,
    maxAdults: 5,
    price: 266,
    goods: [
      `Air conditioning`,
      `Baby seat`,
      `Towels`,
      `Washer`,
      `Breakfast`,
      `Fridge`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    location: {
      latitude: 48.86561,
      longitude: 2.350499,
      zoom: 16
    },
    id: 87
  },
  {
    city: {
      name: `Brussels`,
      location: {
        latitude: 50.846558,
        longitude: 4.351697,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`
    ],
    title: `The house among olive `,
    isFavorite: true,
    isPremium: false,
    rating: 4.6,
    type: `apartment`,
    bedrooms: 4,
    maxAdults: 7,
    price: 218,
    goods: [
      `Washer`,
      `Laptop friendly workspace`,
      `Breakfast`,
      `Air conditioning`,
      `Baby seat`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 50.86556,
      longitude: 4.371697,
      zoom: 16
    },
    id: 3
  },
  {
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938362,
        longitude: 6.959974,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`
    ],
    title: `Nice, cozy, warm big bed apartment`,
    isFavorite: true,
    isPremium: false,
    rating: 2.5,
    type: `house`,
    bedrooms: 2,
    maxAdults: 6,
    price: 485,
    goods: [
      `Breakfast`,
      `Washer`,
      `Baby seat`,
      `Laptop friendly workspace`,
      `Air conditioning`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
    location: {
      latitude: 50.945362,
      longitude: 6.935974,
      zoom: 16
    },
    id: 34
  }
];

export const cities = {
  Amsterdam: {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  },
  Brussels: {
    name: `Brussels`,
    location: {
      latitude: 50.846558,
      longitude: 4.351697,
      zoom: 13
    }
  },
  Cologne: {
    name: `Cologne`,
    location: {
      latitude: 50.938362,
      longitude: 6.959974,
      zoom: 13
    }
  },
  Dusseldorf: {
    name: `Dusseldorf`,
    location: {
      latitude: 51.225403,
      longitude: 6.776314,
      zoom: 13
    }
  },
  Hamburg: {
    name: `Hamburg`,
    location: {
      latitude: 53.550343,
      longitude: 10.000654,
      zoom: 13
    }
  },
  Paris: {
    name: `Paris`,
    location: {
      latitude: 48.85661,
      longitude: 2.351499,
      zoom: 13,
    }
  }
};


export const reviews = [
  {
    id: 18,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `вапвпапвапвавапва вапвпапвапвавапвавапвпапвапвавапвавапвпапвапвавапвавапвпапвапвавапва`,
    date: `2020-11-24T14:44:02.210Z`
  },
  {
    id: 17,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `Test1 To submit review please make sure to setratingand describe your stay with at least50 characters.\nTo submit review please make sure to setratingand describe your stay with at least50 characters.\n\n\n`,
    date: `2020-11-22T01:32:42.268Z`
  },
  {
    id: 16,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 2,
    comment: `234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123`,
    date: `2020-11-22T01:32:15.744Z`
  },
  {
    id: 15,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 3,
    comment: `sadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadass`,
    date: `2020-11-22T01:20:16.707Z`
  },
  {
    id: 14,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `sadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsada`,
    date: `2020-11-22T01:18:54.165Z`
  },
  {
    id: 13,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `efeffqw efeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqw`,
    date: `2020-11-21T16:15:10.563Z`
  },
  {
    id: 12,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 3,
    comment: `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk. `,
    date: `2020-11-21T16:09:20.677Z`
  },
  {
    id: 11,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312`,
    date: `2020-11-21T16:08:28.941Z`
  },
  {
    id: 10,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 2,
    comment: `wqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwe`,
    date: `2020-11-21T16:07:46.202Z`
  },
  {
    id: 9,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `textFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextField`,
    date: `2020-11-21T16:07:00.223Z`
  },
  {
    id: 8,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `wqewqewqewqewqewqeqweqweqwewqewqeqweqweqweqweqweqweqweqweqweqweqweqweqweqwewqeqweqweqweqwewqeqweqw`,
    date: `2020-11-21T16:05:44.050Z`
  },
  {
    id: 7,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 5,
    comment: `sdsfsdfdsfsdfdsfsdfdsf`,
    date: `2020-11-21T15:37:35.160Z`
  },
  {
    id: 6,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 5,
    comment: `qweqweqweqweqweqweqdsdsdsdqwdqwdweqweqweqweqwdwqddwdwq`,
    date: `2020-11-21T14:48:36.841Z`
  },
  {
    id: 5,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 5,
    comment: `qweqweqweqweqweqweqwewqeqwedweweqewqewqewqewqeqwewqwq`,
    date: `2020-11-21T14:47:26.390Z`
  },
  {
    id: 4,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 5,
    comment: `fsfsfsd`,
    date: `2020-11-16T17:26:36.123Z`
  },
  {
    id: 3,
    user: {
      id: 15,
      isPro: false,
      name: `Kendall`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`
    },
    rating: 3,
    comment: `I stayed here for one night and it was an unpleasant experience.`,
    date: `2020-11-01T13:38:44.752Z`
  },
  {
    id: 2,
    user: {
      id: 12,
      isPro: true,
      name: `Isaac`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`
    },
    rating: 3,
    comment: `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
    date: `2020-11-01T13:38:44.752Z`
  },
  {
    id: 1,
    user: {
      id: 14,
      isPro: true,
      name: `Corey`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`
    },
    rating: 5,
    comment: `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    date: `2020-11-01T13:38:44.752Z`
  }
];


export const user = {
  authorizationStatus: `AUTH`,
  data: {
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`,
    email: `r-sasha@list.ru`,
    id: 1,
    isPro: false,
    name: ``
  },
  error: ``
};


export const titleInputs = [
  `terribly`,
  `badly`,
  `not bad`,
  `good`,
  `perfect`
];


export const optionList = {
  FILTER_POPULAR: `Popular`,
  FILTER_TOP_RATED_FIRST: `Top rated first`,
  FILTER_PRICE_LOW_TO_HIGH: `Price: low to high`,
  FILTER_PRICE_HIGH_TO_LOW: `Price: high to low`
};


export const baseFilter = `Popular`;
