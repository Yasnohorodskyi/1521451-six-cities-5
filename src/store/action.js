export const ActionType = {
  CHANGE_CITY: `CHANGE_CITY`,
  GET_OFFERS: `GET_OFFERS`,
  FILTER_OFFER: `FILTER_OFFER`
};

export const ActionCreator = {
  changeCity: (titleCity) => ({
    type: ActionType.CHANGE_CITY,
    payload: {
      title: titleCity
    }
  }),
  filterOffer: (filter, offers) => ({
    type: ActionType.FILTER_OFFER,
    payload: {
      filter,
      offers
    }
  }),
};

