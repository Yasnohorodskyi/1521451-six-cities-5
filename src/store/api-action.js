export const fetch = () => (dispatch, _getState, api) => (
  api.get(`https://5.react.pages.academy/six-cities/hotels`)
    .then(({data}) => dispatch({
      type: `LOAD_OFFERS`,
      payload: data
    }))
);
