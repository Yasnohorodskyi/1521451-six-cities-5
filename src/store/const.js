export const OffersType = {
  CHANGE_CITY: `CHANGE_CITY`,
  GET_OFFERS: `GET_OFFERS`,
  GET_OFFER: `GET_OFFER`,
  REMOVE_FAVORITE: `REMOVE_FAVORITE`,
  FILTER_OFFER: `FILTER_OFFER`,
  REDIRECT_TO_ROUTE: `REDIRECT_TO_ROUTE`,
  GET_REVIEWS: `GET_REVIEWS`,
  ADD_REVIEWS: `ADD_REVIEWS`,
  SET_FAVORITE: `SET_FAVORITE`,
  GET_FAVORITE: `GET_FAVORITE`,
  ERROR_LOAD: `ERROR_LOAD`,
  ERROR_OFFER: `ERROR_OFFER`,
  ERROR_GET_FAVORITE: `ERROR_GET_FAVORITE`,
  ERROR_SET_FAVORITE: `ERROR_SET_FAVORITE`,
};

export const UserType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  RESULT: ``,
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
  REQUIRED_ERROR: `REQUIRED_ERROR`,
};

export const ReviewType = {
  ADD_REVIEWS: `ADD_REVIEWS`,
  GET_REVIEWS: `GET_REVIEWS`,
  ERROR_LOAD: `ERROR_LOAD`,
  ERROR_ADD: `ERROR_ADD`
};

export const AppRoute = {
  RESULT: `/`,
  LOGIN: `/login`
};

const BASE_URL = `https://5.react.pages.academy/six-cities/`;

export const APIRoute = {
  GET_OFFER: BASE_URL + `hotels`,
  GET_REVIEWS: BASE_URL + `comments`,
  LOGIN: BASE_URL + `login`,
  REDIRECT_TO_ROUTE: `REDIRECT_TO_ROUTE`,
  FAVORITE: BASE_URL + `favorite`,
};

export const ActionFilter = {
  FILTER_POPULAR: `Popular`,
  FILTER_TOP_RATED_FIRST: `Top rated first`,
  FILTER_PRICE_LOW_TO_HIGH: `Price: low to high`,
  FILTER_PRICE_HIGH_TO_LOW: `Price: high to low`,
};
