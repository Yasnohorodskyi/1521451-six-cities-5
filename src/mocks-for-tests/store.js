import configureStore from "redux-mock-store";
import {offers, cities, reviews, user, baseFilter} from "./mocks";
import {offersAdapter, offerAdapter} from "../helpers/offers-adapter";

const createStore = configureStore();

const mockStore = createStore({
  Offers: {
    currentCity: cities[`Brussels`],
    baseFilter,
    offers: offersAdapter(offers),
    offer: offerAdapter(offers[0]),
    nearby: offersAdapter(offers),
    favorites: offersAdapter(offers),
    data: offersAdapter(offers),
    getFavoriteDispatch: ()=>{}
  },
  Review: {
    reviews
  },
  User: user,
  Offer: offerAdapter(offers[0])
});

export {mockStore};
