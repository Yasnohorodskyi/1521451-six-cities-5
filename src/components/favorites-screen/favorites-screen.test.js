import React from "react";
import {Provider} from "react-redux";
import {
  MemoryRouter
} from 'react-router-dom';

import {mockStore} from "../../mocks-for-tests/store";
import FavoritesScreen from "./favorites-screen";
import renderer from "react-test-renderer";


describe(`FavoriteScreen`, () => {

  let headerScreenComponent = null;

  beforeEach(() => {

    mockStore.dispatch = jest.fn();

    headerScreenComponent = renderer.create(
        <MemoryRouter>
          <Provider store={mockStore}>
            <FavoritesScreen />
          </Provider>
        </MemoryRouter>
    );
  });

  it(`Should render correctly`, () => {
    expect(headerScreenComponent.toJSON()).toMatchSnapshot();
  });

});
