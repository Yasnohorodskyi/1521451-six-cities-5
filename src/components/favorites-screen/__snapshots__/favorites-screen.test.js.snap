// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteScreen Should render correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-linkheader__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <div
                className="header__nav-link header__nav-link--profile"
              >
                <img
                  className="header__avatar-wrapper"
                  src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg"
                />
                <span
                  className="header__user-name user__name"
                >
                  <a
                    href="/favorites"
                    onClick={[Function]}
                  >
                     
                    r-sasha@list.ru
                     
                  </a>
                </span>
              </div>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Cologne
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/34"
                  >
                    <img
                      alt="Nice, cozy, warm big bed apartment"
                      className="place-card__image"
                      height="110"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        485
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        className="rating__w100"
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/34"
                      onClick={[Function]}
                    >
                      Nice, cozy, warm big bed apartment
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    house
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Brussels
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/3"
                  >
                    <img
                      alt="The house among olive "
                      className="place-card__image"
                      height="110"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        218
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        className="rating__w100"
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/3"
                      onClick={[Function]}
                    >
                      The house among olive 
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    apartment
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Hamburg
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/11"
                  >
                    <img
                      alt="The house among olive "
                      className="place-card__image"
                      height="110"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        190
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        className="rating__w100"
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/11"
                      onClick={[Function]}
                    >
                      The house among olive 
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    house
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Paris
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/87"
                  >
                    <img
                      alt="Tile House"
                      className="place-card__image"
                      height="110"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        266
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        className="rating__w100"
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/87"
                      onClick={[Function]}
                    >
                      Tile House
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    hotel
                  </p>
                </div>
              </article>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
